src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -15,39 +15,153 @@
 #include "deps_format.h"
 
 
+void get_all_fx_versions(
+    host_mode_t mode,
+    const pal::string_t& own_dir,
+    const pal::string_t& fx_name,
+    std::vector<pal::string_t>* fx_versions)
+{
+    // No FX resolution for standalone apps
+    if (mode == host_mode_t::standalone)
+    {
+        trace::verbose(_X("Standalone mode detected. Not gathering shared FX locations"));
+        return;
+    }
+
+    // No FX resolution for mixed apps
+    if (mode == host_mode_t::split_fx)
+    {
+        trace::verbose(_X("Split/FX mode detected. Not gathering shared FX locations"));
+        return;
+    }
+
+    pal::string_t local_dir;
+    std::vector<pal::string_t> global_dirs;
+
+    // own_dir contains DIR_SEPARATOR appended that we need to remove.
+    pal::string_t own_dir_temp = own_dir;
+    remove_trailing_dir_seperator(&own_dir_temp);
+
+    std::vector<pal::string_t> hive_dir;
+    hive_dir.push_back(own_dir_temp);
+
+    if (pal::get_global_dotnet_dirs(&global_dirs))
+    {
+        for (pal::string_t dir : global_dirs)
+        {
+            if (dir != own_dir_temp)
+            {
+                hive_dir.push_back(dir);
+            }
+        }
+    }
+
+    for (pal::string_t dir : hive_dir)
+    {
+        auto fx_dir = dir;
+        trace::verbose(_X("Gathering FX locations in [%s]"), fx_dir.c_str());
+
+        append_path(&fx_dir, _X("shared"));
+        append_path(&fx_dir, fx_name.c_str());
+
+        if (pal::directory_exists(fx_dir))
+        {
+            std::vector<pal::string_t> versions;
+            pal::readdir(fx_dir, &versions);
```

Comment:
Does this also read files named `1.2.3` ?

Reply:
Test

Reply:
Added a new commit for this. Also updated the other runtime code that only needs to query directories, so it doesn't try to look at files or the "." and ".." directories.